### PR TITLE
Dh refactor

### PR DESCRIPTION
**Pull request by mixmix**
this branches off the vectors branch. so merge that first : https://github.com/ssbc/ssb-private-group-keys/pull/5

the purpose of this is to tidy up dh-keys use. I'm refactoring it to be able to more easily weave in the new po-box work

### CODE REVIEW

**Comment by arj03** [comment]
Can you rebase on top of master?

**Comment by arj03** [comment]
Also I'm wondering if we should release a 1.0 version of this library before merging this? The reason being that it makes it easier to see that this library was created "together" with merging in this PR https://github.com/ssbc/ssb-tribes/pull/44.

**Comment by mixmix** [comment]
I can rebase when I'm awake if I get to have some sleep >:(
Lots of screaming here tonight

Any comments on the code / how I've changed it. 

I dunno about versions. I would have thought this module was still in dev, and we are about to release 1.0.0
This is an API change to dh-keys, so if it's in the api you can import need to think about that. 
My preference is to have it in v1, but I don't mind that much

**Comment by arj03** [comment]
No worries, I can review based on the direct commit.

As for the version thing, I wanted to get your feeling on that matter. It's fine by me that we keep working on this repo until we feel  its good.

File: dh-keys.js [hunk 1]
```diff
@@ -3,20 +3,48 @@ const na = require('sodium-native')
 // this is for converting FeedKeys (ed25119 signing keys) into
 // Diffie-Hellman keys (curve25519 based shared key encryption)
 
+// na.randombytes_buf(buf)
```
Comment (arj03) [inline dh-keys.js:6]: This doesn't look right? ;)

File: dh-keys.js [hunk 2]
```diff
@@ -45,4 +73,35 @@ module.exports = class DHKeys {
...
+
+function bufferize (key, length) {
+  if (Buffer.isBuffer(key)) return key
+
+  if (typeof (key) !== 'string') throw new Error(`unable to bufferize ${typeof key}`)
+
+  return Buffer.from(
+    key
+      .replace('.ed25519', '')
+      .replace('.bbfeed-v1', ''),
```
Comment (arj03) [inline dh-keys.js:91]: same as the other commit, we don't need the bbfeed check here

**Comment by arj03** [comment]
I had to read the po box spec to see why you did this change. Overall I think it looks good so far. I would probably add a `generate` method instead of overloading the constructor, that can be a bit of a footgun.

File: dh-keys.js (hunk 1 above)
Comment (mixmix) [reply dh-keys.js:6]: Oh yeah, that was a snippet I was gonna use. Can tidy

File: dh-keys.js (hunk 2 above)
Comment (mixmix) [reply dh-keys.js]: done

**Comment by mixmix** [comment]
@arj03 pulled out generate method. Like your suggestion with that.
Merging this

**Pull request merged**